plugins/c8y_configuration_plugin/src/child_device.rs

```diff
@@ -0,0 +1,216 @@
+            },
+        );
+    }
+}
+
+impl ChildDeviceOperations {
+    const C8Y_STATIC_TEMPLATE_PUBLISH_TOPIC: &'static str = "c8y/s/us";
+
+    pub fn get_child_id(&self) -> String {
+        match self {
+            ChildDeviceOperations::Update { child_id, .. } => child_id.to_string(),
+            ChildDeviceOperations::Snapshot { child_id, .. } => child_id.to_string(),
+        }
+    }
+
+    pub fn get_config_type(&self) -> String {
+        match self {
+            ChildDeviceOperations::Update { file_entry, .. } => file_entry.config_type.to_string(),
+            ChildDeviceOperations::Snapshot { file_entry, .. } => {
+                file_entry.config_type.to_string()
+            }
+        }
+    }
+
+    pub fn get_child_topic(&self) -> String {
+        match self {
+            ChildDeviceOperations::Update { child_id, .. } => {
+                format!("{}/{}", Self::C8Y_STATIC_TEMPLATE_PUBLISH_TOPIC, child_id)
+            }
+            ChildDeviceOperations::Snapshot { child_id, .. } => {
+                format!("{}/{}", Self::C8Y_STATIC_TEMPLATE_PUBLISH_TOPIC, child_id)
+            }
+        }
+    }
+
+    pub fn http_file_repository_relative_path(&self) -> String {
+        match self {
+            ChildDeviceOperations::Update {
+                child_id,
+                file_entry,
+                ..
+            } => {
+                format!("{}/config_update/{}", child_id, file_entry.config_type)
+            }
+            ChildDeviceOperations::Snapshot {
+                child_id,
+                file_entry,
+                ..
+            } => {
+                format!("{}/config_snapshot/{}", child_id, file_entry.config_type)
+            }
+        }
+    }
+
+    /// The configuration management topic for a child device.
+    ///
+    /// # Example:
+    /// For a configuration update returns:
+    ///     - "tedge/CHILD_ID/commands/req/config_update"
+    ///
+    /// For a configuration snapshot returns:
+    ///     - "tedge/CHILD_ID/commands/req/config_snapshot"
+    pub fn child_topic_request(&self) -> Topic {
+        match self {
+            ChildDeviceOperations::Update { child_id, .. } => {
+                Topic::new_unchecked(&format!("tedge/{}/commands/req/config_update", child_id))
+            }
+            ChildDeviceOperations::Snapshot { child_id, .. } => {
+                Topic::new_unchecked(&format!("tedge/{}/commands/req/config_snapshot", child_id))
+            }
+        }
+    }
+
+    /// The configuration management payload for a child device.
+    pub fn request_payload(
+        &self,
+        local_http_host: &str,
+    ) -> Result<String, ChildDeviceConfigManagementError> {
+        match self {
+            ChildDeviceOperations::Update {
+                child_id: _,
+                file_entry,
+            } => {
+                let url = format!(
+                    "http://{local_http_host}/tedge/file-transfer/{}",
```

Comment:
I see that `file-transfer` path is used here, but it's not used elsewhere. Make sure it's consistent with that `file-transfer` path everywhere.